### PR TITLE
Samples of protocol separation by kotlin interface

### PR DESCRIPTION
**Pull request by yuinacor**
Simple samples of protocol separation by kotlin interface.

This includes two different patterns of handing pure kotlin model which declared by interface.

### CODE REVIEW

File: shop/server/src/main/resources/application.yaml [hunk 1]
```diff
@@ -5,3 +5,4 @@ coffee:
 juice:
   api:
     baseUrl: http://localhost:8082
+    user-agent: shop
```
Comment (noveler17) [inline shop/server/src/main/resources/application.yaml]: This config isn't related to PR's goal.

File: juice/api/client/src/main/kotlin/com/linecorp/sample/juice/api/client/JuiceApiClientAutoConfiguration.kt [hunk 2]
```diff
@@ -31,11 +36,17 @@ class JuiceApiClientAutoConfiguration {
                     }
                     .build())
             .baseUrl(properties.baseUrl)
-            .addConverterFactory(JacksonConverterFactory.create(jacksonObjectMapper()))
+            .addConverterFactory(JacksonConverterFactory.create(jacksonObjectMapper().registerTypeModule<Juice, JuiceResponse>()))
             .build()
             .create<JuiceApi>()
 }
 
+inline fun <reified INTERFACE, reified IMPLEMENTATION : INTERFACE> ObjectMapper.registerTypeModule(): ObjectMapper = registerModule(typeModule<INTERFACE, IMPLEMENTATION>())
```
Comment (noveler17) [inline juice/api/client/src/main/kotlin/com/linecorp/sample/juice/api/client/JuiceApiClientAutoConfiguration.kt:44]: :+1:

**Review by noveler17 — APPROVED** [review]
LGTM w/o a nit.

File: shop/server/src/main/resources/application.yaml (hunk 1 above)
Comment (yuinacor) [reply shop/server/src/main/resources/application.yaml]: Fixed

**Pull request merged**